src/localStorage.ts

```diff
@@ -1,26 +1,70 @@
 import { Connectors } from 'use-wallet';
+import { CONNECTORS } from './web3/constants';
 
 const STORAGE_PREFIX = '__mStable-app__';
+const STORAGE_VERSION = 1;
 
-interface Storage {
+interface BaseStorage {
+  version?: number;
+}
+
+interface StorageV0 extends BaseStorage {
   connectorId?: keyof Connectors;
 }
 
-const getStorageKey = (key: keyof Storage): string =>
+interface StorageV1 extends BaseStorage {
+  version: 1;
+  connector?: { id: keyof Connectors; label: string };
+}
+
+type AllStorage = StorageV0 | StorageV1;
+
+export type Storage = Extract<AllStorage, { version: typeof STORAGE_VERSION }>;
+
+const getStorageKey = <S extends BaseStorage>(key: keyof S): string =>
   `${STORAGE_PREFIX}.${key}`;
 
 export const LocalStorage = {
-  set<T extends keyof Storage>(key: T, value: Storage[T]): void {
-    window.localStorage.setItem(getStorageKey(key), JSON.stringify(value));
+  set<S extends BaseStorage, T extends keyof S>(key: T, value: S[T]): void {
+    window.localStorage.setItem(getStorageKey<S>(key), JSON.stringify(value));
   },
-  get<T extends keyof Storage>(key: T): Storage[T] {
-    const value = window.localStorage.getItem(getStorageKey(key));
+  get<S extends BaseStorage, T extends keyof S>(key: T): S[T] {
+    const value = window.localStorage.getItem(getStorageKey<S>(key));
     return value && value.length > 0 ? JSON.parse(value) : undefined;
   },
-  clearItem<T extends keyof Storage>(key: T): void {
-    window.localStorage.setItem(getStorageKey(key), '');
+  removeItem<S extends BaseStorage, T extends keyof S>(key: T): void {
+    window.localStorage.removeItem(getStorageKey<S>(key));
   },
   clearAll(): void {
     window.localStorage.clear();
   },
 };
+
+const migrateLocalStorage = (): void => {
+  const version = LocalStorage.get('version');
+
+  if (version === 0) {
```

Comment:
A migration is needed here because it's now necessary to store the connector `id` and `label`. We can't reliably migrate storage for the `injected` provider unless we assume everyone's using MetaMask, so those users will need to reconnect (just once).

Reply:
No problems. If you are sure it's backwards compatible (reconnect aside) then resolve this convo.